tests/contrib/openai/test_openai_llmobs.py

```diff
@@ -696,6 +698,267 @@ def test_chat_stream_no_resp(self, openai, ddtrace_global_config, mock_llmobs_wr
+                model_name=resp.model,
+                model_provider="openai",
+                input_messages=[{"role": "user", "content": input_messages}],
+                output_messages=response_tool_function_expected_output,
+                metadata={
+                    "temperature": 1.0,
+                    "top_p": 1.0,
+                    "tools": [
+                        {
+                            "type": "function",
+                            "name": "get_current_weather",
+                            "description": "Get the current weather in a given location",
+                            "parameters": {
+                                "type": "object",
+                                "properties": {
+                                    "location": {
+                                        "type": "string",
+                                        "description": "The city and state, e.g. San Francisco, CA",
+                                    },
+                                    "unit": {"type": "string", "enum": ["celsius", "fahrenheit"]},
+                                },
+                                "required": ["location", "unit"],
+                                "strict": True,
+                            },
+                        }
+                    ],
+                    "tool_choice": "auto",
+                    "truncation": "disabled",
+                    "text": {"format": {"type": "text"}},
+                    "reasoning_tokens": 0,
+                },
+                token_metrics={"input_tokens": 75, "output_tokens": 23, "total_tokens": 98},
+                tags={"ml_app": "<ml-app-name>", "service": "tests.contrib.openai"},
+            )
+        )
+
+    @pytest.mark.skipif(
+        parse_version(openai_module.version.VERSION) < (1, 66), reason="Response options only available openai >= 1.66"
+    )
+    def test_response_function_call_stream(self, openai, mock_llmobs_writer, mock_tracer, snapshot_tracer):
+        """Test that Response tool calls are recorded as LLMObs events correctly."""
+        with get_openai_vcr(subdirectory_name="v1").use_cassette("response_function_call_streamed.yaml"):
+            model = "gpt-4.1"
+            input_messages = "What is the weather like in Boston today?"
+            client = openai.OpenAI()
+            resp = client.responses.create(
+                tools=response_tool_function,
+                model=model,
+                input=input_messages,
+                user="ddtrace-test",
+                stream=True,
+            )
+            for chunk in resp:
+                if hasattr(chunk, "response") and hasattr(chunk.response, "model"):
+                    resp_model = chunk.response.model
+        span = mock_tracer.pop_traces()[0][0]
+        assert mock_llmobs_writer.enqueue.call_count == 1
+        response_tool_function_expected_output[0]["tool_calls"][0]["tool_id"] = "call_lGe2JKQEBSP15opZ3KfxtEUC"
```

Comment:
Instead of overwriting the constant, let's just create a copy and then overwrite that copy
```
streamed_response_tool_expected_output = response_tool_function_expected_output.copy()
streamed_response_tool_expected_output[0]["..."] = "..."
```